scripts/check-sidebar-lazy-layout.py

```
@@ -1,741 +1,657 @@
-                "Sidebar row bodies must be value-only so AppKit lifecycle "
-                "callbacks cannot mutate row state during SwiftUI layout "
-                "(issue #8004).".format(region_name, name)
-            )
+    body = extract_function_body(clean, "workspaceScrollArea")
+    if body is None:
+        return ["could not locate func workspaceScrollArea(...); update this guard for the renamed boundary"]
+    if "CmuxFeatureFlags.shared.isAppKitSidebarListEnabled" not in body:
+        violations.append("workspaceScrollArea does not preserve the AppKit-sidebar rollout flag")
+    for helper in ("appKitWorkspaceScrollArea", "legacyWorkspaceScrollArea"):
+        if not re.search(r"\b" + helper + r"\s*\(", body):
+            violations.append(f"workspaceScrollArea does not route through {helper}")
```

Ruff (0.15.21)

[warning] Use `list.extend` to create a transformed list

(PERF401)